includes/Core/Admin_Bar/Admin_Bar.php

```diff
@@ -134,41 +134,55 @@ private function add_menu_button( $wp_admin_bar ) {
 	 * @return bool True if Admin bar should display, False when it's not.
 	 */
 	public function is_active() {
-		if ( ! is_user_logged_in() || ! is_admin_bar_showing() ) {
+
+		// Only active if the admin bar is showing.
+		if ( ! is_admin_bar_showing() ) {
 			return false;
 		}
 
-		// Gets post object. On front area we need to use get_queried_object to get the current post object.
-		if ( $this->is_admin_post_screen() ) {
-			$post = get_post();
-		} else {
-			$post = get_queried_object();
-		}
+		// Determine the queried object.
+		$queried_object = get_queried_object();
 
-		if ( ! $post || ! $post instanceof \WP_Post ) {
-			return false;
-		}
+		// Checks for post objects.
+		if ( $queried_object instanceof \WP_Post ) {
 
-		if ( 'publish' !== $post->post_status ) {
-			return false;
+			// Ensure the user can view post insights for this post.
+			if ( ! current_user_can( Permissions::VIEW_POST_INSIGHTS, $queried_object->ID ) ) {
+				return false;
+			}
+
+			// Only published posts show the menu.
+			if ( 'publish' !== $queried_object->post_status ) {
+				return false;
+			}
+		} else {
+
+			// Only admins can see non post admin bar data.
+			if ( ! current_user_can( Permissions::MANAGE_OPTIONS ) ) {
+				return false;
+			}
 		}
 
-		if ( ! current_user_can( Permissions::VIEW_POST_INSIGHTS, $post->ID ) ) {
-			return false;
+		// Data is based on the current URL.
+		if ( $this->is_admin_post_screen() ) {
+			global $post;
+			$current_url = $this->context->get_reference_permalink( $post->ID );
+		} else {
+			$current_url = $this->context->get_reference_canonical();
 		}
 
 		/**
 		 * Filters whether the Site Kit admin bar menu should be displayed.
 		 *
-		 * The admin bar menu is only intended for when a single published post is being visited, hence this filter is fired only
-		 * under these circumstances.
+		 * The admin bar menu is only shown when there is data for the current URL.
 		 *
 		 * @since 1.0.0
 		 *
-		 * @param bool $display Whether to display the admin bar menu.
-		 * @param int  $post_id Currently visited post ID.
+		 * @param bool   $display     Whether to display the admin bar menu.
+		 * @param string $current_url The URL of the current request.
+		 * @param mixed  $post_id     Currently queried object if available.
 		 */
-		return apply_filters( 'googlesitekit_show_admin_bar_menu', true, (int) $post->ID );
+		return apply_filters( 'googlesitekit_show_admin_bar_menu', true, $current_url, $queried_object );
```

Comment:
Contrary to what is defined in the ACs, I now actually think we shouldn't pass the post ID here anymore. Whether admin bar stats exist or not entirely depends on the URL. Given that the post can also be `null`, having it as a parameter would be weird anyway.

Reply:
I came to the same conclusion while working on this. We really only need the post ID to determine the user's specific level of access to that post (eg a post author might get to see the menu). This is not module/data related, so doesn't belong in the filter.

Reply:
Removed in https://github.com/google/site-kit-wp/pull/543/commits/2e303bf7bc534523ce1636febdfbc346fd9058a9. Should I update the AC as well?